test/xds/xds_client_ignore_resource_deletion_test.go

```diff
@@ -0,0 +1,474 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package xds_test
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/internal"
+	"google.golang.org/grpc/internal/testutils"
+	"google.golang.org/grpc/internal/testutils/xds/bootstrap"
+	"google.golang.org/grpc/internal/testutils/xds/e2e"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/xds"
+
+	clusterpb "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	corepb "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	endpointpb "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+	listenerpb "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	routepb "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	testServicepb "google.golang.org/grpc/test/grpc_testing"
+	testpb "google.golang.org/grpc/test/grpc_testing"
+)
+
+const (
+	serviceName = "my-service-xds"
+	rdsName     = "route-" + serviceName
+	cdsName1    = "cluster1-" + serviceName
+	cdsName2    = "cluster2-" + serviceName
+	edsName1    = "eds1-" + serviceName
+	edsName2    = "eds2-" + serviceName
+)
+
+var (
+	resolverBuilder                 = internal.NewXDSResolverWithConfigForTesting.(func([]byte) (resolver.Builder, error))
+	nodeID                          = uuid.New().String()
```

Comment:
I'm a little concerned about using the same nodeID across tests. Would prefer if we can keep the tests as hermetic as possible.

Reply:
The reason for this being global is to avoid passing it along to all the helper function and since this was a string constant. I think maybe i should have not used many nested helpers then. 

Do you think this is a problem now?

Reply:
I've run into problems in the past when the same nodeID was reused across tests, but maybe at that point we were also using a single management server instance across tests. But since this is simple cleanup of declaring the nodeID inside the test and passing it whatever helper needs it, I think we should do it. Because if it ever leads to some test flakes, it will be hard to debug.

Reply:
Ill update this one in a bit